### PR TITLE
Track patches

### PR DESCRIPTION
**Pull request by sakibguy**

### CODE REVIEW

**Review by sakib-rahman-bangladesh — COMMENTED** [review]
goin gto die. 10k+ changes. eed cool brain b4 analysis

File: drivers/soundwire/Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ soundwire-cadence-y := cadence_master.o
 obj-$(CONFIG_SOUNDWIRE_CADENCE) += soundwire-cadence.o
 
 #Intel driver
-soundwire-intel-y :=	intel.o intel_init.o
+soundwire-intel-y :=	intel.o intel_init.o dmi-quirks.o
```
Comment (sakibguy) [inline drivers/soundwire/Makefile]: updated / renamed intel driver obj file

File: drivers/soundwire/bus.h [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #define DEFAULT_BANK_SWITCH_TIMEOUT 3000
 #define DEFAULT_PROBE_TIMEOUT       2000
 
+u64 sdw_dmi_override_adr(struct sdw_bus *bus, u64 addr);
```
Comment (sakibguy) [inline drivers/soundwire/bus.h]: wired u64 type

File: drivers/soundwire/dmi-quirks.c [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+#include <linux/device.h>
+#include <linux/dmi.h>
+#include <linux/soundwire/sdw.h>
+#include "bus.h"
+
+struct adr_remap {
+	u64 adr;
+	u64 remapped_adr;
+};
```
Comment (sakibguy) [inline drivers/soundwire/dmi-quirks.c]: new type adr_remap

File: drivers/soundwire/dmi-quirks.c [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	{
+		0x000010025D070100,
+		0x000020025D071100
+	},
+	{
+		0x000110025d070100,
+		0x000120025D130800
+	},
+	{}
+};
```
Comment (sakibguy) [inline drivers/soundwire/dmi-quirks.c]: 4hp device 
static type adr_remap hp_spectre_360

File: drivers/soundwire/slave.c [hunk 5]
```diff
@@ -95,7 +95,7 @@ static bool find_slave(struct sdw_bus *bus,
 		       struct acpi_device *adev,
 		       struct sdw_slave_id *id)
 {
-	unsigned long long addr;
+	u64 addr;
```
Comment (sakibguy) [inline drivers/soundwire/slave.c]: changed 
unsigned long long 
to 
custom defined u64 
type addr

File: drivers/soundwire/intel.c [hunk 6]
```diff
@@ -1302,6 +1302,7 @@ static int intel_prop_read(struct sdw_bus *bus)
 
 static struct sdw_master_ops sdw_intel_ops = {
 	.read_prop = sdw_master_read_prop,
+	.override_adr = sdw_dmi_override_adr,
```
Comment (sakibguy) [inline drivers/soundwire/intel.c]: init new var

File: include/linux/soundwire/sdw.h [hunk 7]
```diff
@@ -813,7 +814,8 @@ struct sdw_defer {
  */
 struct sdw_master_ops {
 	int (*read_prop)(struct sdw_bus *bus);
-
+	u64 (*override_adr)
+			(struct sdw_bus *bus, u64 addr);
```
Comment (sakibguy) [inline include/linux/soundwire/sdw.h]: added 
u64 struct pointer

**Review by sakib-rahman-bangladesh — COMMENTED** [review]
new changes 
with 
new custom types

**Review by sakib-rahman-bangladesh — COMMENTED** [review]
renamed

**Pull request closed without merging**